core/src/sumeragi/network_topology.rs

```diff
@@ -391,7 +391,7 @@ impl Topology {
         signatures: impl IntoIterator<Item = &'a SignatureOf<VersionedValidBlock>> + 'a,
     ) -> Vec<SignatureOf<VersionedValidBlock>> {
         let roles: HashSet<Role> = roles.iter().copied().collect();
-        let public_keys: Vec<_> = roles
+        let public_keys: HashSet<_> = roles
```

Comment:
Is this irrelevant?

Reply:
Sure, you can do this for clarity

Reply:
not just for clarity, this collection is searched for an item in a loop so should be better to have set instead of a vec